myapp/spec/system/tasks_spec.rb

```diff
@@ -2,32 +2,47 @@
   describe 'index page' do 
-    it 'show page title, list and delete buttons' do 
-      Task.create!(title: 'Test title 1', description: 'Test Description 1', due: Time.zone.now + 5, status: 'pending')
-      Task.create!(title: 'Test title 2', description: 'Test Description 2', due: Time.zone.now + 5, status: 'pending')
-
-      visit tasks_path
-
-      # page title
-      expect(page).to have_content('Task List')
-
-      # filter field
-      ## search title
-      expect(page).to have_field('Search Title:')
-      ## search status
-      expect(page).to have_select('Search Status:')
-      ## sort by
-      expect(page).to have_select('Sort by')
-      ## sort order
-      expect(page).to have_select('Order')
-      ## filter button
-      expect(page).to have_button('Filter')
-
-      # tasks' title
-      expect(page).to have_content('Test title 1')
-      expect(page).to have_content('Test title 2')
+    context 'render all components' do 
+      it 'show page title, list and delete buttons' do 
+        Task.create!(title: 'Test title 1', description: 'Test Description 1', due: Time.zone.now + 5, status: 'pending')
+        Task.create!(title: 'Test title 2', description: 'Test Description 2', due: Time.zone.now + 5, status: 'pending')
+  
+        visit tasks_path
+  
+        # page title
+        expect(page).to have_content('Task List')
+  
+        # filter field
+        ## search title
+        expect(page).to have_field('Search Title:')
+        ## search status
+        expect(page).to have_select('Search Status:')
+        ## sort by
+        expect(page).to have_select('Sort by')
+        ## sort order
+        expect(page).to have_select('Order')
+        ## filter button
+        expect(page).to have_button('Filter')
+  
+        # tasks' title
+        expect(page).to have_content('Test title 1')
+        expect(page).to have_content('Test title 2')
+  
+        expect(page).to have_button('Delete', count: 2)
+      end
+
+      it 'show pagination bar' do 
+        Task.create!(title: 'Test title 1', description: 'Test Description 1', due: Time.zone.now + 5, status: 'pending')
+        Task.create!(title: 'Test title 2', description: 'Test Description 2', due: Time.zone.now + 5, status: 'pending')
+        Task.create!(title: 'Test title 3', description: 'Test Description 3', due: Time.zone.now + 5, status: 'pending')
+        Task.create!(title: 'Test title 4', description: 'Test Description 4', due: Time.zone.now + 5, status: 'pending')
+        Task.create!(title: 'Test title 5', description: 'Test Description 5', due: Time.zone.now + 5, status: 'pending')
+        Task.create!(title: 'Test title 6', description: 'Test Description 6', due: Time.zone.now + 5, status: 'pending')
+
+        visit tasks_path
 
-      expect(page).to have_button('Delete', count: 2)
+        expect(page).to have_selector("nav[class='pagination']")
+      end
     end
 
     context 'interact with buttons' do
```

Comment:
[NITS]
`context` would be specific condition so that code will be more readable.
https://qiita.com/uchiko/items/d34c5d1298934252f58f

How about this?
```suggestion
    context 'when there are some tasks' do
```